.github/workflows/ci_image_build.yml

```diff
@@ -0,0 +1,56 @@
+name: Build CI Image
+
+on:
+  pull_request:
+    paths:
+      - 'docker/ci/**'
+      - '.github/workflows/ci_image_build.yml'
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
```

Comment:
@JLockerman Is this the place that we should be explicit about version of ubuntu that we want to stick to rather than the latest?

Reply:
@jerryxwu 🤔 this is only ever used to build the docker image, so I'm not sure how much it matters either way

Reply:
@jerryxwu I'll pin `20.04` under the assumption that it's slightly more likely to keep working that way